shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-rewrite/src/test/resources/sharding/select.xml

```diff
@@ -453,4 +453,19 @@
         <input sql="SELECT * FROM t_account, t_account_detail WHERE t_account.account_id = 1 AND t_account.account_id = t_account_detail.account_id" />
         <output sql="SELECT * FROM t_account_1, t_account_detail_1 WHERE t_account_1.account_id = 1 AND t_account_1.account_id = t_account_detail_1.account_id" />
     </rewrite-assertion>
+
+    <rewrite-assertion id="select_with_join_table_on_condition_with_qualified_columns">
```

Comment:
Hi, this PR looks generally correct,  thanks for your work.
Also, the focus of the relevant issue is about `whereSegment` and SQL route, right? So could you add an integrated test to make sure this `join and in` SQL has a correct execution result completely?

### How?
Please refer to the case in line 102 of `shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/cases/dql/dql-integration-test-cases.xml`. This case is very similar to your PR.

And last but not least, have a great holiday, :)

Reply:
OK. I will add any case for this PR.